includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -127,17 +145,41 @@ public function get_scripts() {
 		return $scripts;
 	}
 
+	/**
+	 * Process alternative names in attribute spec list.
+	 *
+	 * @since 0.7
+	 *
+	 * @param array $attr_spec_list Attribute spec list.
+	 * @return array Modified attribute spec list.
+	 */
+	private function process_alternate_names( $attr_spec_list ) {
+		foreach ( $attr_spec_list as $attr_name => &$attr_spec ) {
+			if ( '[' === $attr_name[0] ) {
```

Comment:
Yes, since such attributes aren't understood by `DOMDocument` we add an safe placeholder version of that same attribute (e.g. `amp-binding-1234…-src`) and add it to the alternative names so it can be matched instead when the nodes are processed.

Reply:
Perhaps we could use another term than `binding` in the future if some of the attributes are not specific to the binding component.